### PR TITLE
CI: Remove ASAN job

### PR DESCRIPTION
**Pull request by WillAyd**

### CODE REVIEW

**Comment by WillAyd** [comment]
I think for now can just remove this until we can root cause it

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -68,14 +68,6 @@ jobs:
           - name: "Pyarrow Nightly"
             env_file: actions-311-pyarrownightly.yaml
             pattern: "not slow and not network and not single_cpu"
-          - name: "ASAN / UBSAN"
-            env_file: actions-311-sanitizers.yaml
-            pattern: "not slow and not network and not single_cpu and not skip_ubsan"
-            asan_options: "ASAN_OPTIONS=detect_leaks=0"
```
Comment (mroeschke) [inline .github/workflows/unit-tests.yml:74]: Could you remove these extra GHA args as well that are only used for this job too?

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -68,14 +68,6 @@ jobs:
           - name: "Pyarrow Nightly"
             env_file: actions-311-pyarrownightly.yaml
             pattern: "not slow and not network and not single_cpu"
-          - name: "ASAN / UBSAN"
-            env_file: actions-311-sanitizers.yaml
```
Comment (datapythonista) [inline .github/workflows/unit-tests.yml:72]: I guess we can remove this file too?

**Comment by mroeschke** [comment]
If you want to come back to this at a later time to debug and not completely remove some of the configurations, you could do:


```diff
--- a/.github/workflows/unit-tests.yml
+++ b/.github/workflows/unit-tests.yml
@@ -160,7 +160,7 @@ jobs:
 
     - name: Test (not single_cpu)
       uses: ./.github/actions/run-tests
-      if: ${{ matrix.name != 'Pypy' }}
+      if: ${{ matrix.name != 'Pypy' || matrix.name != 'ASAN / UBSAN'}}
       with:
         preload: ${{ matrix.preload }}
         asan_options: ${{ matrix.asan_options }}
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @WillAyd

**Pull request merged**